crates/resolver/src/name_server/name_server.rs

```diff
@@ -19,59 +20,53 @@ use proto::multicast::MDNS_IPV4;
 use proto::xfer::{DnsHandle, DnsRequest, DnsResponse, FirstAnswer};
 use tracing::debug;
 
-#[cfg(feature = "mdns")]
-use crate::config::Protocol;
 use crate::config::{NameServerConfig, ResolverOpts};
 use crate::error::ResolveError;
-use crate::name_server::{ConnectionProvider, NameServerState, NameServerStats};
-#[cfg(feature = "tokio-runtime")]
-use crate::name_server::{TokioConnection, TokioConnectionProvider, TokioHandle};
+use crate::name_server::{GenericConnection, NameServerState, NameServerStats, RuntimeProvider};
+#[cfg(feature = "mdns")]
+use proto::multicast::{MdnsClientConnect, MdnsClientStream, MdnsQueryType};
 
-/// Specifies the details of a remote NameServer used for lookups
+/// This struct is needed only for testing. Specifically, `C` is needed for mocking.
 #[derive(Clone)]
-pub struct NameServer<
-    C: DnsHandle<Error = ResolveError> + Send,
-    P: ConnectionProvider<Conn = C> + Send,
+pub struct AbstractNameServer<
+    C: DnsHandle<Error = ResolveError> + Send + Sync + 'static + CreateConnection,
+    P: RuntimeProvider + Send,
```

Comment:
`RuntimeProvider` already requires `Send`, so this trait bound is no longer needed?

Reply:
You are right. I will resolve it.